### PR TITLE
Try: Create REST API for task list

### PR DESCRIPTION
**Pull request by joshuatf**
This PR is an experiment in creating a REST API for the task list.  It is incomplete as is, but kicks off this process by:

* Moving tasks to a REST endpoint
* Refactoring task lists to use SlotFill
* Removing preloaded options

I'd like to get an early review before continuing further and to verify that this is in fact the route we want to go.

### Why move to a REST API?

* **Extensibility** - the current task list is hard to extend, even for our own tasks we create obscure workarounds with onClick where SlotFill would be more appropriate
* **Single source of truth** - currently checking to see if tasks are complete or what the status of a task list is has become nightmare-ish.  No blame on anyone here, it's just grown into something unwieldy as we add more and more features.  Some statuses are preloaded from the server, some are fetched via getOption , some are retrieved from a status endpoint.  This is very hard to parse and definitely has some circular logic.
* **Performance** - Retrieving all data from a central REST endpoint will probably improve load times given we don't have to preload everything and all data can be retrieved from a single endpoint.

### Disadvantages

The primary disadvantage to this, in my opinion, is that we'll need to invalidate or re-fetch the task data during certain events.

For example: with the previous client-side completion checks, an installed plugin would immediately be reflected in the task list with this logic `isComplete: activePlugins.includes( 'my-plugin' )`.

We will now need to subscribe to certain events and invalidate cache accordingly if we want those same benefits.

### Follow-up tasks 

* Add snooze endpoint and `snoozeUntil` properties on tasks via filter
* Add dismiss endpoint and `isDismissed` properties on tasks via filter
* Data store (for task retrieval, dismissal, and snooze)
* Data store cache invalidation
* Clean up around extended task list
* Removal of `tasks/status` endpoint

### Screenshots

Screen Shot 2021-07-14 at 3 57 29 PM

### Detailed test instructions:

1. Navigate to the task list
2. Play around with tasks

Note that a data store is needed to correctly update completion status and a few tasks may not be fully complete.

### CODE REVIEW

**Comment by mattsherman** [comment]
@joshuatf Since we now have the #6856 epic, can we close this PR?

**Comment by joshuatf** [comment]
We can indeed, @mattsherman.
